userspace/libsinsp/examples/test.cpp

```diff
@@ -339,12 +352,33 @@ void plaintext_dump(sinsp& inspector)
 	}
 }
 
+void json_dump_init(sinsp& inspector)
+{
+	if (!json_dump_init_success && !json_dump_reinit_fmt_success)
```

Comment:
I don't think checking for `json_dump_reinit_fmt_success` is needed here, since it can only be `true` if `json_dump_reinit_evt_formatter` has been called at least once and that function also checks `json_dump_init` has been called once so `json_dump_init_success` will already be true. Put in other terms, there will be no scenario where `json_dump_reinit_fmt_success` will be true and `json_dump_init_success` will be false.

If that's the case, I think we can completely remove `json_dump_reinit_fmt_success` and simplify the logic a bit.

Reply:
lol changed it @Molter73 , ty. Work for too long in security and you become paranoid checking things that can and should not happen, because ...